src/parameterizations/lateral/MOM_internal_tides.F90

```diff
@@ -813,16 +842,13 @@ subroutine refract(En, cn, freq, dt, G, US, NAngle, use_PPMang)
                    (G%CoriolisBu(I,J-1) + G%CoriolisBu(I-1,J)))
       df_dx = 0.5*((G%CoriolisBu(I,J) + G%CoriolisBu(I,J-1)) - &
                     (G%CoriolisBu(I-1,J) + G%CoriolisBu(I-1,J-1))) * G%IdxT(i,j)
-      dlnCn_dx = 0.5*( G%IdxCu(I,j) * (cn(i+1,j) - cn(i,j)) / &
-                       (0.5*(cn(i+1,j) + cn(i,j)) + cn_subRO) + &
-                       G%IdxCu(I-1,j) * (cn(i,j) - cn(i-1,j)) / &
-                       (0.5*(cn(i,j) + cn(i-1,j)) + cn_subRO) )
+      dlnCn_dx = G%IdxT(i,j) * (cn_u(I,j) - cn_u(I-1,j)) / (cn(i,j) + cn_subRO)
```

Comment:
Even with the cn_subR0, using just a single value for cn(i,j) in the denominator could be dangerously large if that value is 0 (or very small) but one of the adjacent values of cn_u in the numerator is non-zero. Perhaps it might make sense to replace the denominator with `(0.5*cn(i,j) + 0.25*(cn(i-1,j) + cn(i+1,j)) + cn_subRO)`, and similarly for the denominator of dlnCn_dy below.

Reply:
I think this would re-introduce the problematic values from the halos or land, instead I propose to test for values of cn

Reply:
The updated code addresses this issue very nicely.